### DISCUSSION TITLE
Problem with MIDI out

### DISCUSSION THREAD

**Discussion by bloiiing** [hydrogen-music/hydrogen#1532] [Hydrogen on Windows]
Hello,

I have a problem using hydrogen. I have a 4x4 (4in/4out) MIDI adapter and only the in1 input works. The in2,3,4 and out1,2,3,4 do not work. I don't know if it's a hydrogen bug or if it's a configuration problem? With other software this device works fine. The device is recognized by hydrogen but does not work. No MIDI signal comes out of the software.. I'm on Windows 10.

Thank you in advance.

**Comment by cme** [comment]
This might be caused by issue #1518, can I ask what the names of the midi devices hydrogen sees are?

**Reply by cme** [reply]
If you would like to try a debug build with this issue fixed to check if this is the cause, you can find some here:

64 bit:

https://ci.appveyor.com/api/buildjobs/hov07uww8v026m5t/artifacts/Hydrogen-1.2.0-win64.exe

32 bit:

https://ci.appveyor.com/api/buildjobs/lwxoyp1qq8tihv7j/artifacts/Hydrogen-1.2.0-win32.exe

**Reply by bloiiing** [reply]
> This might be caused by issue #1518, can I ask what the names of the midi devices hydrogen sees are?

This is the midiface 4x4 from miditech. I have tried the 1.2.0 version and the problem remains the same. I have just done an install with and without uninstalling. In both case it doesn't works. But I noticed that when uninstalling some data like old pattern remains when I launch the new version. Perhaps the problem is due to a bad uninstall. I don't know how to uninstall completely?

**Reply by cme** [reply]
I don't think a clean uninstall will help, but if you want to try, you should remove `\Users\<username>\.hydrogen\` from your startup disk.

That debug build also has extra debug information which might help us work out what's going on -- would you be able to post the debug log here, please? Start Hydrogen, try and connect your MIDI devices, and then go to "Debug -> Open log file" to open the log file in Notepad (or other default text editor), then drag that logfile here.

Thanks!

**Reply by bloiiing** [reply]
Here is the log file. In1,2,3,4 works and Out,2,3,4 does't works
[hydrogen.log](https://github.com/hydrogen-music/hydrogen/files/8244171/hydrogen.log)
.

**Reply by bloiiing** [reply]
Sorry, out 1,2,3,4 doesn't works.

**Reply by cme** [reply]
Thanks! That's helpful. From the looks of this logfile, Hydrogen successfully found and opened the output device, but there wasn't enough debug tracing to tell whether there was anything sent to the output device or not. I'll add some more tracing and like the updated build here for you to try, if you don't mind giving it a go!

Your cooperation is super-useful, since so many of these things are specific to hardware or platform dependencies that we just don't have access to in order to test ourselves. Thanks!

**Reply by bloiiing** [reply]
I don't mind trying if it can help you. Thank you for taking the time to resolve this issue. Goodbye!

**Reply by cme** [reply]
64-bit installer: 
https://ci.appveyor.com/api/buildjobs/9m48e25vt61mwwod/artifacts/Hydrogen-1.2.0-win64.exe

**Reply by bloiiing** [reply]
Here is the log file. Thanks.
[hydrogen.log](https://github.com/hydrogen-music/hydrogen/files/8249803/hydrogen.log)
.

**Reply by cme** [reply]
Thanks! From the looks of this, it looks like the problem is just that your drumkit doesn't have MIDI out channel assignments for the instruments.

http://hydrogen-music.org/documentation/manual/manual_en.html#chpt.instrument_editor.midi_out_settings

I think this is a pretty clunky bit of interface design in that
  1) the shipped kits should probably have a default out channel, or at least a warning if there's a MIDI output configured but no instruments have a MIDI out channel, and 
  2) having to set each instrument's output channel individually is a pain since the most common case is probably to set all the instruments to the same channel.

I'm going to open an issue to that effect!

**Reply by bloiiing** [reply]
It works. Thank you very much. It was just a settings problem. Not a bug. All my excuses.

**Comment by bloiiing** [comment]
This is the midiface 4x4 device of miditech. Names of inputs are the same that outputs.. The driver is portmidi. I'm going to try with only one input or output to see if it works. I will let you know the result. Thank you for your answer.
